core/src/main/java/org/elasticsearch/common/xcontent/json/JsonXContent.java

```diff
@@ -46,11 +46,14 @@ public static XContentBuilder contentBuilder() throws IOException {
     }
 
     private final static JsonFactory jsonFactory;
+    public final static String JSON_ALLOW_UNQUOTED_FIELD_NAMES = "elasticsearch.json.allow_unquoted_field_names";
     public final static JsonXContent jsonXContent;
 
     static {
         jsonFactory = new JsonFactory();
-        jsonFactory.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES, false);
+        // TODO: Remove the system property configuration for this in Elasticsearch 6.0.0
+        jsonFactory.configure(JsonParser.Feature.ALLOW_UNQUOTED_FIELD_NAMES,
+                "true".equals(System.getProperty(JSON_ALLOW_UNQUOTED_FIELD_NAMES)) ? true : false);
```

Comment:
"tru" parsing to `false` is less leniency, the value must be "true" and ONLY "true" to set this option. All other values are false